### PR TITLE
create Node.js Custom Plugin Template

### PR DESCRIPTION
**Pull request by Luca-Ferrera**

### CODE REVIEW

File: config/Dockerimage [hunk 1]
```diff
@@ -0,0 +1 @@
+%NEXUS_HOSTNAME%/%CUSTOM_PLUGIN_IMAGE_NAME%
```
Comment (fredmaggiowski) [inline config/Dockerimage]: This file is uselss

File: index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+/* eslint require-await: 0 */
+'use strict'
+
+const customService = require('@mia-platform/custom-plugin-lib')({
+})
+
+module.exports = customService(async function index(service) {
+
+  /*
+  * Insert here your code
```
Comment (fredmaggiowski) [inline index.js]:
```suggestion
  * Insert your code here.
```

File: tests/index.test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    BACKOFFICE_HEADER_KEY: 'backoffice',
+    MICROSERVICE_GATEWAY_SERVICE_NAME: 'microservice-gateway.example.org',
+  })
+
+  t.tearDown(async() => {
+    await fastify.close()
+  })
+
+  /*
+  * Insert here your test
```
Comment (fredmaggiowski) [inline tests/index.test.js]:
```suggestion
  * Insert your tests here.
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+Once you have all your dependency in place you can launch:
+```shell
+set -a && source .env
+npm start
+```
+
+After that you will have the service exposed on your machine.
+
+## Configuring the Service
+<TO COMPLETE>
```
Comment (davidebianchi) [inline README.md]: Could we complete? 😄

File: index.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint require-await: 0 */
+'use strict'
+
+const customService = require('@mia-platform/custom-plugin-lib')({
+})
```
Comment (davidebianchi) [inline index.js]: Close the brackets inline?

File: package.json [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    "lint": "eslint . --ignore-path=.gitignore",
+    "start": "lc39 index.js",
+    "start:local": "npm start -- --env-path ./default.env",
+    "test": "npm run lint && npm run unit && npm run checkonly",
+    "update-docker-version": "sed -i.bck \"s|version=\\\"[0-9]*\\.[0-9]*\\.[0-9]*.*\\\"|version=\\\"${npm_package_version}\\\"|\" Dockerfile",
+    "unit": "tap -b tests/*.test.js",
+    "version": "npm run update-docker-version && rm -fr Dockerfile.bck && git add Dockerfile"
+  },
+  "dependencies": {
+    "@mia-platform/custom-plugin-lib": "^1.0.5"
```
Comment (davidebianchi) [inline package.json]: Can we upgrade the deps?

**Comment by davidebianchi** [comment]
LGTM

@fredmaggiowski could you take a look?

File: tests/index.test.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    BACKOFFICE_HEADER_KEY: 'backoffice',
+    MICROSERVICE_GATEWAY_SERVICE_NAME: 'microservice-gateway.example.org',
+  })
+
+  t.tearDown(async() => {
+    await fastify.close()
+  })
+
+  /*
+  * Insert your tests here.
```
Comment (fredmaggiowski) [inline tests/index.test.js]:
```suggestion
   * Insert your tests here.
```

File: tests/index.test.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  */
```
Comment (fredmaggiowski) [inline tests/index.test.js]:
```suggestion
   */
```

**Pull request merged**